packages/ui/src/containers/SocialSignIn/SecondarySocialSignIn.test.tsx

```diff
@@ -36,19 +38,28 @@ describe('SecondarySocialSignIn', () => {
   });
 
   it('less than four connectors', () => {
-    const { container } = render(
-      <MemoryRouter>
-        <SecondarySocialSignIn connectors={socialConnectors.slice(0, 3)} />
-      </MemoryRouter>
+    const { container } = renderWithPageContext(
+      <SettingsProvider
+        settings={{
+          ...mockSignInExperienceSettings,
+          socialConnectors: socialConnectors.slice(0, 3),
```

Comment:
nit: I've seen `socialConnectors.length > 3` and `socialConnectors.slice(0, 3)` numerous times. Can we somehow define a global variable for the `3` and use it everywhere?
Or maybe we can put this logic in the `useSocial` hook? Any thoughts?

Reply:
make sense let me check

Reply:
Update. Add it at the component level as it controls the UI display logic only and may differ between different components.